go.mod

```diff
@@ -2,7 +2,9 @@ module github.com/containers/podman/v5
 
 // Warning: Ensure the "go" and "toolchain" versions match exactly to prevent unwanted auto-updates
 
-go 1.22.0
+go 1.22.6
+
+toolchain go1.22.7
```

Comment:
This is wrong, we do not want to force newer toolchains for no reason see the comment above.

Reply:
You cannot revert the `go` parts as this is hard enforced by go module now to be the minimum version used for builds. So if any dependency is using this we can either not bump the dep or have to live with that.

What we do control is the toolchain line and for this one we should never use it as the only purpose would be to force a newer version (compared to the go line) to compile with.

Reply:
unfortunately that doesn't solve the problem with centos-stream 9/10.

The updated `go` part is coming from containers/image.  I either drop the c/image bump patch or we accept these failures for now.

Reply:
Well I brought that up when we did the update to go 1.22, I consider bumping to a z version as minimum fundamentally broken behavior that simply should not happen in a stable language but go decided to allow this so now we are forced to live with that if we want to consume dependencies that just force "random" go updates and yes this means we will be unable to build in slower updating envs. I don't know how much of a problem this will be in practise but I guess if we cannot build on RHEL or even fedora we have a rather big problem.

ref https://github.com/containers/image/pull/2550#discussion_r1742435929
and we decided in standup that we are ok to bump z versions.